src/main/kotlin/no/nav/tilleggsstonader/sak/behandling/vent/SettPåVentService.kt

```diff
@@ -0,0 +1,164 @@
+package no.nav.tilleggsstonader.sak.behandling.vent
+
+import no.nav.tilleggsstonader.kontrakter.oppgave.Oppgave
+import no.nav.tilleggsstonader.sak.behandling.BehandlingService
+import no.nav.tilleggsstonader.sak.behandling.domain.Behandling
+import no.nav.tilleggsstonader.sak.behandling.domain.BehandlingStatus
+import no.nav.tilleggsstonader.sak.behandling.historikk.BehandlingshistorikkService
+import no.nav.tilleggsstonader.sak.behandling.historikk.domain.StegUtfall
+import no.nav.tilleggsstonader.sak.infrastruktur.exception.feilHvis
+import no.nav.tilleggsstonader.sak.infrastruktur.sikkerhet.SikkerhetContext
+import no.nav.tilleggsstonader.sak.opplysninger.oppgave.OppgaveService
+import org.springframework.stereotype.Service
+import org.springframework.transaction.annotation.Transactional
+import java.time.LocalDate
+import java.util.Optional
+import java.util.UUID
+
+@Service
+class SettPåVentService(
+    private val behandlingService: BehandlingService,
+    private val behandlingshistorikkService: BehandlingshistorikkService,
+    private val oppgaveService: OppgaveService,
+    private val settPåVentRepository: SettPåVentRepository,
+) {
+
+    fun hentSettPåVent(behandlingId: UUID): StatusPåVentDto {
+        val settPåVent = settPåVentRepository.findByBehandlingIdAndAktivIsTrue(behandlingId)
+            ?: error("Finner ikke settPåVent for behandling=$behandlingId")
+        val oppgave = oppgaveService.hentOppgave(settPåVent.oppgaveId)
+
+        return StatusPåVentDto(
+            årsaker = settPåVent.årsaker,
+            kommentar = settPåVent.kommentar,
+            frist = oppgave.fristFerdigstillelse,
+            oppgaveVersjon = oppgave.versjonEllerFeil(),
+        )
+    }
+
+    @Transactional
+    fun settPåVent(behandlingId: UUID, dto: SettPåVentDto): StatusPåVentDto {
+        val behandling = behandlingService.hentBehandling(behandlingId)
+        feilHvis(behandling.status.behandlingErLåstForVidereRedigering()) {
+            "Kan ikke sette behandling på vent når status=${behandling.status}"
+        }
+        opprettHistorikkInnslag(behandling, StegUtfall.SATT_PÅ_VENT, mapOf("årsaker" to dto.årsaker))
+        behandlingService.oppdaterStatusPåBehandling(behandlingId, BehandlingStatus.SATT_PÅ_VENT)
+
+        val oppgave = hentOppgave(behandlingId)
+        // TODO må bruke enhet til saksbehandler, men 4462 er enheten til NAY. Enhetene til tiltak er uklart
+        val mappeId = oppgaveService.finnMapper("4462").single { it.navn == "10 På vent" }.id.toLong()
```

Comment:
Dette så litt hacky ut, men er kanskje sånn det må gjøres? 😢

Reply:
Det er slik jeg tolket det, og det er sånn det gjøres i EF

Reply:
For no er det så og sei den einaste måten å gjere dette på, men når me får inn tiltaksenhetene så blir det meir problemstillinger som kjem inn i denne kodesnutten.